src/app/index/index.component.ts

```diff
@@ -16,7 +16,7 @@ export class IndexComponent implements OnInit {
 
   };
   constructor(private route: ActivatedRoute, private router: Router, private store: Store<fromRoot.State> ) {
-    this.store.dispatch(new queryactions.QueryAction(''));
+    this.store.dispatch(new queryactions.QueryServerAction(''));
```

Comment:
basically, that should be both actions getting dispatched. please dispatch QueryAction along with QueryServerAction.